build/templates/_attributes.py.mako

```diff
@@ -104,6 +105,38 @@ class AttributeEnum(object):
         return self._underlying_attribute.__set__(session, value.value)
 
 
+class AttributeEnumWithConverter(AttributeEnum):
+    '''Class for attributes that use enums internally but are exposed in nimi-python API as something else, thus need conversion.'''
+
+    def __init__(self, underlying_enum_attribute, getter_converter, setter_converter):
```

Comment:
since the name of the class you pass in as `underlying_enum_attribute` is `AttributeEnum`, consider changing variable name to `underlying_attribute_enum`

Reply:
I have updated this in my latest commit.